// twilio-run/dist/commands/shared.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sharedCliOptions = exports.sharedApiRelatedCliOptions = exports.baseCliOptions = void 0;
const logger_1 = require("../utils/logger");
exports.baseCliOptions = {
    logLevel: {
        type: 'string',
        default: 'info',
        alias: 'l',
        describe: 'Level of logging messages.',
        choices: Object.keys(logger_1.LoggingLevel),
    },
};
exports.sharedApiRelatedCliOptions = {
    region: {
        type: 'string',
        hidden: true,
        describe: 'Twilio API Region',
    },
    edge: {
        type: 'string',
        hidden: true,
        describe: 'Twilio API Region',
    },
    'account-sid': {
        type: 'string',
        alias: 'u',
        describe: 'A specific account SID to be used for deployment. Uses fields in .env otherwise',
    },
    'auth-token': {
        type: 'string',
        describe: 'Use a specific auth token for deployment. Uses fields from .env otherwise',
    },
    'load-system-env': {
        default: false,
        type: 'boolean',
        describe: 'Uses system environment variables as fallback for variables specified in your .env file. Needs to be used with --env explicitly specified.',
    },
};
exports.sharedCliOptions = Object.assign(Object.assign({}, exports.baseCliOptions), { config: {
        alias: 'c',
        type: 'string',
        default: '.twilio-functions',
        describe: 'Location of the config file. Absolute path or relative to current working directory (cwd)',
    }, cwd: {
        type: 'string',
        describe: 'Sets the directory of your existing Serverless project. Defaults to current directory',
    } });
